common/Util.cpp

```diff
@@ -114,7 +118,12 @@ namespace Util
             std::vector<char> v(length);
 
             int len = 0;
-#ifdef HAVE_SYS_RANDOM_H
+#ifdef HAVE_GETENTROPY
+            len = getentropy(v.data(), length);
+
+            // if getentropy() fails, try "/dev/[u]random"
+            if (len == -1)
```

Comment:
I find the if condition and body being separated by macro checks (and thus possibly not even being a conditional block) a bit dangerous. Not sure what the best way to handle it would be, I would perhaps add a bool variable for the if condition that has a default value true, and the code inside the #ifdef/#elif sets it according to the condition.

Reply:
Done.
Thanks.